changelog.d/pr-7247.md

```diff
@@ -0,0 +1,8 @@
+### 🚀 Enhancements and New Features
+
+- Support for annex private mode.
+  A new `--reckless private` mode was added to the `clone`, `get` & `install` commands; a new `--private` option was added to `create`.
+  Using private mode will configure the new clone with `annex.private=true`, meaning that the clone won't store any information about itself in the git-annex branch (the branch will still exist and contain information about other clones).
+  This mode can be used for temporary clones, where changes are pushed (e.g. back to origin), and the temporary clone is promptly discarded.
```

Comment:
- Ideally we would like to harmonize how `private` is specified across commands.  But not yet sure if `--private` or `--reckless`:
- For `reckless` -- can't it be a combination of `shared-` and `private` or `ephemeral` and `private` or some other  in particular `wreckless` is we do want to quickly do some change in a clone to push back?

Reply:
> Ideally we would like to harmonize how private is specified across commands. But not yet sure if --private or --reckless

I'm trying to remember what was the reason I chose the current way, but can't. Probably just following the suggestion in #6456 to reuse the existing switch for clone, and then not feeling comfortable with the calling the create operation "reckless".

> For reckless -- can't it be a combination of shared- and private or ephemeral and private or some other in particular wreckless is we do want to quickly do some change in a clone to push back?

Umm... For ephemeral & private: that wouldn't change anything because clone ephemeral already sets annex.private (which is the only DataLad usage of annex.private outside this PR). For shared & private, and yet-to-be-invented and private: I did not think of it, but I suppose it would technically be possible...

Reply:
FWIW, I agree that it is useful to reuse the ``--reckless`` switch, especially when the combination ``--reckless ephemeral --private`` becomes possible but doesn't make sense. And I agree it makes sense to call it ``--private`` instead of reckless for ``create``